src/main/java/org/openrewrite/java/template/processor/RefasterTemplateProcessor.java

```diff
@@ -735,7 +735,7 @@ public RuleDescriptor(JCTree.JCClassDecl classDecl, JCCompilationUnit cu, Contex
 
             if (classDecl.typarams != null && !classDecl.typarams.isEmpty()) {
                 printNoteOnce("Generic type parameters are currently not supported", classDecl.sym);
-                return null;
+                //TODO return null;
```

Comment:
We'll need to figure out in which exact cases we can tolerate the generic types and still end up with working recipes.

Reply:
AFAICT we can use this approach for any unbound type parameters (so basically `T` or `T extends Object`), but as soon as there are bounds this would go wrong. But I think this would already represent an improvement.

We would also have to make sure to not generate anything for templates using the type parameter directly for a parameter like `boolean before(Collection<T> c, T t) { return c.contains(t); }`, as I suspect that this would not work properly.